src/lib/transition_handler/breadcrumb_builder.ml

```diff
@@ -90,24 +90,42 @@ module Make (Inputs : Inputs.S) :
                            breadcrumb_if_present ()
                          in
                          new_breadcrumb)
-                  | Error (`Fatal_error exn) ->
-                      Deferred.return (Or_error.of_exn exn)
-                  | Error (`Invalid_staged_ledger_hash error) ->
-                      let%map () =
-                        Trust_system.record_envelope_sender trust_system logger
-                          sender
-                          ( Trust_system.Actions.Gossiped_invalid_transition
-                          , Some ("invalid staged ledger hash", []) )
+                  | Error err -> (
+                      (* propagate bans through subtree *)
+                      let subtrees = Rose_tree.flatten subtree in
+                      let ip_address_set =
+                        let sender_from_tree_node node =
+                          Envelope.Incoming.sender (Cached.peek node)
+                        in
+                        List.fold subtrees
+                          ~init:(Set.empty (module Unix.Inet_addr))
+                          ~f:(fun inet_addrs node ->
+                            match sender_from_tree_node node with
+                            | Local ->
+                                inet_addrs
```

Comment:
We may want to throw an error in this case, maybe even crash. This state would indicate something is *very* wrong, if I'm not mistaken.